contrib/views/utils/src/main/java/org/apache/ambari/view/utils/hdfs/ConfigurationBuilder.java

```diff
@@ -243,24 +250,62 @@ private void copyHAProperties(String defaultFS) throws URISyntaxException, HdfsA
       HA_NAMENODES_INSTANCE_PROPERTY);
 
     String[] namenodes = namenodeIDs.split(",");
-    if (namenodes.length != 2) {
-      throw new HdfsApiException("HDFS080 " + HA_NAMENODES_INSTANCE_PROPERTY + " namenodes count is not exactly 2");
+    //    get the property values from cluster.
+    //    If not found then get the values from view instance property.
+
+    List<String> rpcAddresses = new ArrayList<>(namenodes.length);
+    List<String> httpAddresses = new ArrayList<>(namenodes.length);
+    List<String> httpsAddresses = new ArrayList<>(namenodes.length);
+    for (String namenode : namenodes) {
+
+      String rpcAddress = getProperty(HDFS_SITE, String.format(NAMENODE_RPC_NN_CLUSTER_PROPERTY, nameservice, namenode));
+      if(!Strings.isNullOrEmpty(rpcAddress)) {
+        rpcAddresses.add(rpcAddress);
+      }
+
+      String httpAddress = getProperty(HDFS_SITE, String.format(NAMENODE_HTTP_NN_CLUSTER_PROPERTY, nameservice, namenode));
+      if(!Strings.isNullOrEmpty(httpAddress)) {
+        httpAddresses.add(httpAddress);
+      }
+
+      String httpsAddress = getProperty(HDFS_SITE, String.format(NAMENODE_HTTPS_NN_CLUSTER_PROPERTY, nameservice, namenode));
+      if(!Strings.isNullOrEmpty(httpsAddress)) {
+        httpsAddresses.add(httpsAddress);
+      }
+    }
+
+    if(rpcAddresses.isEmpty()){
+      //      get property from view instance configs
+      String viewRpcAddresses = getViewProperty(NAMENODE_RPC_NN_INSTANCE_PROPERTY);
+      LOG.debug("value of {} in view is : {}", NAMENODE_RPC_NN_INSTANCE_PROPERTY, viewRpcAddresses);
+      if(!Strings.isNullOrEmpty(viewRpcAddresses)){
+        rpcAddresses.addAll(Arrays.asList(viewRpcAddresses.split(",")));
+      }
+    }
+
+    if(httpAddresses.isEmpty()){
+      //      get property from view instance configs
+      String viewHttpAddresses = getViewProperty(NAMENODE_HTTP_NN_INSTANCE_PROPERTY);
+      LOG.debug("value of {} in view is : {}", NAMENODE_HTTP_NN_INSTANCE_PROPERTY, viewHttpAddresses);
+      if(!Strings.isNullOrEmpty(viewHttpAddresses)){
+        httpAddresses.addAll(Arrays.asList(viewHttpAddresses.split(",")));
+      }
+    }
+
+    if(httpsAddresses.isEmpty()){
+    //      get property from view instance configs
+      String viewHttpsAddresses = getViewProperty(NAMENODE_HTTPS_NN_INSTANCE_PROPERTY);
+      LOG.debug("value of {} in view is : {}", NAMENODE_HTTPS_NN_INSTANCE_PROPERTY, viewHttpsAddresses);
+      if(!Strings.isNullOrEmpty(viewHttpsAddresses)){
+        httpsAddresses.addAll(Arrays.asList(viewHttpsAddresses.split(",")));
+      }
+    }
```

Comment:
Would be nice to extract these almost identical repeated code blocks to methods.